File: src/main/java/moe/giga/discord/commands/PingCommand.java
package moe.giga.discord.commands;

import moe.giga.discord.annotations.CommandInfo;
import moe.giga.discord.contexts.MessageContext;
import net.dv8tion.jda.core.entities.Message;

import java.time.Duration;
import java.time.Instant;

@CommandInfo(name = "ping", description = "Ping command")
public final class PingCommand extends Command {

    @Override
    public void onCommand(MessageContext MC, String[] args) {
        Instant start = Instant.now();
        Message m = MC.sendMessage("Pong!");
        Instant end = Instant.now();
        long delta = Duration.between(start, end).toMillis();

        m.editMessage(String.format("Pong! %dms", delta)).complete();
    }
}

File: src/main/kotlin/moe/giga/discord/commands/PingCommand.kt
package moe.giga.discord.commands

import moe.giga.discord.contexts.MessageContext
import java.time.Duration
import java.time.Instant

@Suppress("unused")
class PingCommand : Command {
    override val name = "ping"
    override val description = "Ping command"

    override fun execute(MC: MessageContext, args: List<String>) {
        val start = Instant.now()
        MC.channel.sendMessage("Pong!").queue {
            val end = Instant.now()
            val delta = Duration.between(start, end).toMillis()
            it.editMessage(String.format("Pong! %dms", delta)).queue()
        }
    }
}
